File: abc/abc344/b.java
// https://atcoder.jp/contests/abc344/submissions/51111026

import java.util.Scanner;

class Main {
  public static void main(String args[]) {
    Scanner stdIn = new Scanner(System.in);

    int[] ans = new int[100];
    for(int i = 0; i < 100; i++) {
      ans[i] = stdIn.nextInt();
      
      if(ans[i] == 0) {
        for(int j = i; j >= 0; j--) {
          System.out.println(ans[j]);
        }
        break;
      }
    }
  }
}

File: abc/abc344/b.cs
// https://atcoder.jp/contests/abc344/submissions/51110919

using static System.Console;

public class A
{
  static void Main()
  {
    List<int> ans = new();
    int x;
    do {
      x = int.Parse(ReadLine());
      ans.Add(x);
    } while(x != 0);

    for(int i = ans.Count() - 1; i >= 0; i--) {
      WriteLine(ans[i]);
    }
  }
}

